lib/std/builtin.zig

```diff
@@ -58,24 +58,24 @@ pub const StackTrace = struct {
     index: usize,
     instruction_addresses: []usize,
 
-    pub fn format(
-        self: StackTrace,
-        comptime fmt: []const u8,
-        options: std.fmt.FormatOptions,
-        writer: anytype,
-    ) !void {
-        var arena = std.heap.ArenaAllocator.init(std.heap.page_allocator);
-        defer arena.deinit();
-        const debug_info = std.debug.getSelfDebugInfo() catch |err| {
-            return writer.print("\nUnable to print stack trace: Unable to open debug info: {s}\n", .{@errorName(err)});
-        };
-        const tty_config = std.debug.detectTTYConfig();
-        try writer.writeAll("\n");
-        std.debug.writeStackTrace(self, writer, &arena.allocator, debug_info, tty_config) catch |err| {
-            try writer.print("Unable to print stack trace: {s}\n", .{@errorName(err)});
-        };
-        try writer.writeAll("\n");
+    pub fn withTTYConfig(
+        self: @This(),
+        tty_config: std.debug.TTY.Config,
+    ) StackTraceWithTTYConfig {
+        return .{ .trace = self, .tty_config = tty_config };
     }
+
+
+    // TODO(rgreenblatt): maybe this shouldn't exist so the default formatter can be used?
+    // TODO(rgreenblatt): should this instead default to using no color?
+    pub const format = @compileError("use 'withTTYConfig' to specify if colors can be used");
+};
+
+pub const StackTraceWithTTYConfig = struct {
+    trace: StackTrace,
+    tty_config: std.debug.TTY.Config,
+
+    pub const format = std.debug.formatStackTraceWithTTYConfig;
```

Comment:
I never liked this default formatter in `builtin.zig` doing so many things at once.
What about nuking it and providing a `std.fmt.Formatter` for it in `debug.zig` ?

Reply:
I added a `fmtStackTrace` function to debug.zig.